New procedure argument metadata option 'defaulted'

gacha opened the pull request:
Now it is possible to determine if a field has default arguments.

Reciepe chef/centos-6.6 doesn't exists any more, so updated it to 6.7.
New CentOS image doesn't permit password from dictionary, so changed it
to something else.

jgebal commented:
Nice work.
Can you add a unit test for the defaulted columns?

gacha commented:
will do

javornikolov commented on lib/plsql/procedure.rb:
```diff
@@ -104,7 +104,7 @@ def get_argument_metadata #:nodoc:
         "SELECT #{subprogram_id_column}, object_name, TO_NUMBER(overload), argument_name, position, data_level,
               data_type, in_out, data_length, data_precision, data_scale, char_used,
-              char_length, type_owner, type_name, type_subname
+              char_length, type_owner, type_name, type_subname, defaulted
```
Since which Oracle release do we have this `defaulted` attribute? What would happen if we run against e.g. 10g?

gacha replied:
Yes, on 10g there's no such column, so I will do like in `subprogram_id_column`.

javornikolov commented:
@gacha : would you mind splitting the test VM setup changes into a separate pull request (seems an independent topic). Thank you!

gacha commented:
Finally got time to finish this, removed the Vagrant change because someone already did that, also added tests.

Also I had problems running tests as it's described in Readme, I had to add `DATABASE_USE_TNS=NO` to anything happen.

javornikolov commented on spec/plsql/procedure_spec.rb:
```diff
@@ -2251,3 +2251,49 @@ def new_candidate(status)
+describe '#get_argument_metadata' do
+  before do
+    plsql.execute <<-SQL
+      CREATE OR REPLACE FUNCTION hr.magic_number(p_num INTEGER #{defaulted_value})
```
Do we need to specify the schema name here `hr.`?

javornikolov replied:
I'm wondering about a better name - `defaulted_value` sounds more like the value itself (21), while we expect the whole expression/clause here.

javornikolov commented on spec/plsql/procedure_spec.rb:
```diff
@@ -2251,3 +2251,49 @@ def new_candidate(status)
 end
+
+describe '#get_argument_metadata' do
+  before do
```
I would vote for being explicit about the [`before`/`after` scope](https://www.relishapp.com/rspec/rspec-core/v/3-4/docs/hooks/before-and-after-hooks).

javornikolov commented:
Thank you @gacha, it looks great! I just posted a few cosmetic remarks which I think is good to sort out and we can merge the PR.

gacha commented:
Anything else?

javornikolov commented:
Thanks, it looks good to me :+1:  @gacha, could you please just squash the two commits into a single one before we merge it.

gacha commented:
Done

javornikolov commented:
Nice, I'm merging it... @gacha, thank you very much for your contribution!
